jgrapht-guava/src/main/java/org/jgrapht/graph/BaseNetworkAdapter.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * (C) Copyright 2018-2018, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+import org.jgrapht.EdgeFactory;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphType;
+import org.jgrapht.graph.AbstractGraph;
+import org.jgrapht.graph.DefaultGraphType;
+
+import com.google.common.graph.Network;
+
+/**
+ * A base abstract implementation for the graph adapter class using Guava's {@link Network}. This is
+ * a helper class in order to support both mutable and immutable networks.
+ * 
+ * @author Dimitrios Michail
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * @param <N> type of the underlying Guava's network
+ */
+public abstract class BaseNetworkAdapter<V, E, N extends Network<V, E>>
+    extends AbstractGraph<V, E>
+    implements Graph<V, E>, Cloneable, Serializable
+{
+    private static final long serialVersionUID = -6742507788742087708L;
+
+    protected static final String LOOPS_NOT_ALLOWED = "loops not allowed";
+
+    protected transient Set<V> unmodifiableVertexSet = null;
+    protected transient Set<E> unmodifiableEdgeSet = null;
+
+    protected EdgeFactory<V, E> edgeFactory;
+    protected N network;
+
+    /**
+     * Create a new network adapter.
+     * 
+     * @param network the mutable network
+     * @param ef the edge factory of the new graph
+     */
+    public BaseNetworkAdapter(N network, EdgeFactory<V, E> ef)
+    {
+        this.edgeFactory = Objects.requireNonNull(ef);
+        this.network = Objects.requireNonNull(network);
+    }
+
+    @Override
+    public E getEdge(V sourceVertex, V targetVertex)
+    {
+        return network
+            .edgesConnecting(sourceVertex, targetVertex).stream().findFirst().orElse(null);
```

Comment:
This is a case of incompatibility between the two interfaces. Guava throws an exception if there are more than one edges (in case of a multi-graph), while JGraphT returns the first edge.

Reply:
Oh okay, that makes sense! Thanks for the clarification. :)